src/common/helpers.ts

```diff
@@ -8,15 +8,14 @@ import { values } from '@azure-tools/linq';
 
 // aggregates the Parameter in op.parameters and the first request
 export function aggregateParameters(op: Operation): Array<Parameter> {
-  if (op.requests!.length > 1) {
-    throw console.error('multiple requests NYI');
-  }
   let params = new Array<Parameter>();
   if (op.parameters) {
     params = params.concat(op.parameters);
   }
-  if (op.requests![0].parameters) {
-    params = params.concat(op.requests![0].parameters);
+  for (const req of values(op.requests)) {
```

Comment:
I'm surprised this doesn't cause unnecessary params.  If there's some magic that happens during transformation that prevents it you should add a comment here explaining it.

Reply:
I dont think this should be producing unnecessary params. In the case of multiple requests, those operations will be separated out after processing the initial operation. For most cases there will only be one request, therefore it will only return the parameters of that request. I can add a comment explaining that in the case of multiple requests the only time this will aggregate parameters for all requests is for the transform and that the operation with multiple requests will later be removed.

Reply:
I see.  Does it need to aggregate params during transformation or is it safe to leave it as `op.requests![0]` then?

Reply:
aggregateParameters is also used in the namer before we get to the transform, so it is necessary to parse all of the parameters for the operation.